src/software/geom/algorithms/end_in_obstacle_sample.h

```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+#include "software/ai/navigator/obstacle/obstacle.hpp"
+#include "software/geom/point.h"
+#include "software/geom/rectangle.h"
+
+std::optional<Point> endInObstacleSample(const std::vector<ObstaclePtr> obstacles,
+                                         const Point& point,
+                                         const Rectangle& navigable_area,
+                                         int initial_count = 6, double rad_step = 0.15,
+                                         int per_rad_step = 2, double range = 2.0);
```

Comment:
I feel like these parameter names are kind of unclear. I personally kept thinking `rad` was referring to radians and not radius. 
Also what is `initial_count` and `range`? Should it be `max_search_radius`? `initial_count` doesn't seem to be used.

Reply:
I renamed the arguments to be clearer

I use initial_count as the initial value for number of points to sample for the first iteration (so 6 points sampled around the initial radius which is set to .15)

Then for each iteration, the number of samples per radius increases by per_rad_step and the radius increases by rad_step (so basically the circumference increases and the number of samples also increases so the density of samples does not drop off too much)

in retrospect i think it might be okay if I just always set the initial_count value to 6, so we don't need to pass it in as an extra argument